content/docs/act.md

```diff
@@ -0,0 +1,109 @@
+---
+id: act
+title: act
+permalink: docs/act.html
+---
+
+`act()` provides an abstraction to make sure what's rendered reflects a state where pending actions have been 'committed' and the DOM has been updated. When writing UI tests, tasks like rendering, user events, or data fetching can be considered as 'units' of interaction with a user interface. `act` allows you be sure all effects related to a unit have been processed before asserting on the result.
+
+Note: The name `act` comes from the `act` section of the [arrange-act-assert pattern](http://wiki.c2.com/?ArrangeActAssert). 
+
+For example:
+
+```jsx
+import { render } from "react-dom";
+import { act } from "react-dom/test-utils";
+import App from "./app";
+
+it("should render content", () => {
+  const container = document.createElement("div");
+  act(() => {
+    render(<App />, container);
+  });
+  // ... make assertions about the UI
+});
+```
+
+### Guarantees {#guarantees}
+
+`act()` provides some guarantees when it's run.
+
+- It batches updates triggered inside its scope, and applies them all in one go.
+- It flushes any queued effects and updates.
+- When nested, the work is flushed only on exiting the outermost `act()`.
+- The asynchronous version of `act()` also flushes any resolved promises.
```

Comment:
This implies the user knows what "batches", "flushes", "effects" mean. I don't think we introduce all of these terms explicitly. Can probably rewrite this in a more concrete way.